### PR TITLE
Add modal/popup for projects

### PR DESCRIPTION
**Pull request by alonsofl**
### Summary
Add modal/popup feature when clicking on the "See Project" button, for mobile and desktop versions.
> Using [Template #3](https://www.figma.com/file/l7SqJ3ZfkAKih9sFxvWSR4/Microverse-Student-Project-1?node-id=1%3A468)

#### Feature Details
- Add HTML skeleton dynamically using JS.
- Add style for the modal/popup in mobile and desktop versions.
- Add Event Listener to open/close the modal/popup.

### CODE REVIEW

File: scripts/main.js [hunk 1]
```diff
@@ -1,3 +1,5 @@
+// Mobile Menu
```
Comment (vichuge) [inline scripts/main.js:1]: - [ ] A good practice on code, is to delete all the commented lines, I kindly invite you to delete it. 🐶

File: scripts/main.js [hunk 2]
```diff
@@ -14,4 +16,137 @@ menuButton.forEach((item) => {
 
 mobileMenuItem.forEach((item) => {
   item.addEventListener('click', toggleMenu);
-});
\ No newline at end of file
+});
+
+// Modal
```
Comment (vichuge) [inline scripts/main.js:21]: - [ ] A good practice on code, is to delete all the commented lines, I kindly invite you to delete it. 🐶

File: scripts/main.js [hunk 3]
```diff
@@ -14,4 +16,137 @@ menuButton.forEach((item) => {
...
+
+const projectButton = document.querySelectorAll('.projects-description-more');
+
+// Create project images array
```
Comment (vichuge) [inline scripts/main.js:25]: - [ ] A good practice on code, is to delete all the commented lines, I kindly invite you to delete it. 🐶

File: scripts/main.js [hunk 4]
```diff
@@ -14,4 +16,137 @@ menuButton.forEach((item) => {
...
+const projectImage = [];
+document.querySelectorAll('.projects-item > img').forEach((item) => {
+  projectImage.push(item.getAttribute('src'));
+});
+
+// Create project titles array
```
Comment (vichuge) [inline scripts/main.js:31]: - [ ] A good practice on code, is to delete all the commented lines, I kindly invite you to delete it. 🐶

File: scripts/main.js [hunk 5]
```diff
@@ -14,4 +16,137 @@ menuButton.forEach((item) => {
...
+const projectTitle = [];
+document.querySelectorAll('.projects-description-title').forEach((item) => {
+  projectTitle.push(item.textContent);
+});
+
+const projectTechnologies = ['HTML', 'CSS', 'JavaScript', 'Ruby On Rails', 'GitHub'];
+
+const projectDescription = 'Lorem ipsum, dolor sit amet consectetur adipisicing elit. Aperiam nostrum consequuntur sint ratione maiores provident veritatis quas a alias atque doloribus quo inventore repudiandae consectetur eius excepturi enim quisquam!';
+
+// Update projects object with information
```
Comment (vichuge) [inline scripts/main.js:41]: - [ ] A good practice on code, is to delete all the commented lines, I kindly invite you to delete it. 🐶

File: scripts/main.js [hunk 6]
```diff
@@ -14,4 +16,137 @@ menuButton.forEach((item) => {
...
+const projects = {};
+for (let i = 0; i < projectImage.length; i += 1) {
+  projects[`project${i}`] = {};
+  projects[`project${i}`].image = projectImage[i];
+  projects[`project${i}`].title = projectTitle[i];
+  projects[`project${i}`].technologies = projectTechnologies;
+  projects[`project${i}`].description = projectDescription;
+}
+
+// Create HTML elements
```
Comment (vichuge) [inline scripts/main.js:51]: - [ ] A good practice on code, is to delete all the commented lines, I kindly invite you to delete it. 🐶

File: scripts/main.js [hunk 7]
```diff
@@ -14,4 +16,137 @@ menuButton.forEach((item) => {
...
+
+// Add section for modal
```
Comment (vichuge) [inline scripts/main.js:53]: - [ ] A good practice on code, is to delete all the commented lines, I kindly invite you to delete it. 🐶

File: scripts/main.js [hunk 8]
```diff
@@ -14,4 +16,137 @@ menuButton.forEach((item) => {
...
+const modalSection = document.createElement('section');
+modalSection.className = 'modal-init';
+document.body.appendChild(modalSection);
+
+// Add closing icon
```
Comment (vichuge) [inline scripts/main.js:58]: - [ ] A good practice on code, is to delete all the commented lines, I kindly invite you to delete it. 🐶

File: scripts/main.js [hunk 9]
```diff
@@ -14,4 +16,137 @@ menuButton.forEach((item) => {
...
+const closeModal = document.createElement('img');
+closeModal.className = 'modal-close-img';
+closeModal.alt = 'Close modal icon';
+closeModal.src = './images/close-mobile-modal.svg';
+modalSection.appendChild(closeModal);
+
+// Add image for project modal
```
Comment (vichuge) [inline scripts/main.js:65]: - [ ] A good practice on code, is to delete all the commented lines, I kindly invite you to delete it. 🐶

File: scripts/main.js [hunk 10]
```diff
@@ -14,4 +16,137 @@ menuButton.forEach((item) => {
...
+const modalImg = document.createElement('img');
+modalImg.className = 'modal-img';
+modalImg.alt = 'Project snapshot';
+modalImg.src = projects.project0.image;
+modalSection.appendChild(modalImg);
+
+// Add div for project description
```
Comment (vichuge) [inline scripts/main.js:72]: - [ ] A good practice on code, is to delete all the commented lines, I kindly invite you to delete it. 🐶

File: scripts/main.js [hunk 11]
```diff
@@ -14,4 +16,137 @@ menuButton.forEach((item) => {
...
+
+const modalDescriptionLeft = document.createElement('div');
+modalDescriptionLeft.className = 'modal-description-left';
+modalDescription.appendChild(modalDescriptionLeft);
+
+const modalDescriptionRight = document.createElement('div');
+modalDescriptionRight.className = 'modal-description-right';
+modalDescription.appendChild(modalDescriptionRight);
+
+// Add content to the left
```
Comment (vichuge) [inline scripts/main.js:85]: - [ ] A good practice on code, is to delete all the commented lines, I kindly invite you to delete it. 🐶

File: scripts/main.js [hunk 12]
```diff
@@ -14,4 +16,137 @@ menuButton.forEach((item) => {
...
+const modalTitle = document.createElement('h3');
+modalTitle.className = 'modal-title';
+modalTitle.textContent = projects.project0.title;
+modalDescriptionLeft.appendChild(modalTitle);
+
+// Add list of technologies
```
Comment (vichuge) [inline scripts/main.js:91]: - [ ] A good practice on code, is to delete all the commented lines, I kindly invite you to delete it. 🐶

File: scripts/main.js [hunk 13]
```diff
@@ -14,4 +16,137 @@ menuButton.forEach((item) => {
...
+let listItemModal = document.createElement('li');
+
+for (let i = 0; i < 5; i += 1) {
+  listItemModal = document.createElement('li');
+  listItemModal.className = 'modal-technologies-item';
+  listItemModal.textContent = projects.project0.technologies[i];
+  listModal.appendChild(listItemModal);
+}
+
+// Add content to the right
```
Comment (vichuge) [inline scripts/main.js:105]: - [ ] A good practice on code, is to delete all the commented lines, I kindly invite you to delete it. 🐶

File: scripts/main.js [hunk 14]
```diff
@@ -14,4 +16,137 @@ menuButton.forEach((item) => {
...
+let buttonModal = document.createElement('li');
+buttonModal.className = 'modal-btn-desktop';
+buttonModal.innerHTML = '<a href="https://alonsofl.github.io/dev-portfolio"> See Live <img src="./images/see-live-icon.svg" alt="See live icon"></a>';
+listDesktopButtonModal.appendChild(buttonModal);
+buttonModal = document.createElement('li');
+buttonModal.className = 'modal-btn-desktop';
+buttonModal.innerHTML = '<a href="https://github.com/alonsofl/dev-portfolio"> See Source <img src="./images/see-source-icon.svg" alt="See source icon"></a>';
+listDesktopButtonModal.appendChild(buttonModal);
+
+// Supporting text
```
Comment (vichuge) [inline scripts/main.js:119]: - [ ] A good practice on code, is to delete all the commented lines, I kindly invite you to delete it. 🐶

File: scripts/main.js [hunk 15]
```diff
@@ -14,4 +16,137 @@ menuButton.forEach((item) => {
...
+const paraModal = document.createElement('p');
+paraModal.className = 'modal-supporting-text';
+paraModal.textContent = projects.project0.description;
+modalSection.appendChild(paraModal);
+
+// Buttons
```
Comment (vichuge) [inline scripts/main.js:125]: - [ ] A good practice on code, is to delete all the commented lines, I kindly invite you to delete it. 🐶

File: scripts/main.js [hunk 16]
```diff
@@ -14,4 +16,137 @@ menuButton.forEach((item) => {
...
+buttonModal = document.createElement('li');
+buttonModal.className = 'modal-btn';
+buttonModal.innerHTML = '<a href="https://alonsofl.github.io/dev-portfolio"> See Live <img src="./images/see-live-icon.svg" alt="See live icon"></a>';
+listButtonModal.appendChild(buttonModal);
+buttonModal = document.createElement('li');
+buttonModal.className = 'modal-btn';
+buttonModal.innerHTML = '<a href="https://github.com/alonsofl/dev-portfolio"> See Source <img src="./images/see-source-icon.svg" alt="See source icon"></a>';
+listButtonModal.appendChild(buttonModal);
+
+// Function to open/close modal
```
Comment (vichuge) [inline scripts/main.js:139]: - [ ] A good practice on code, is to delete all the commented lines, I kindly invite you to delete it. 🐶

File: scripts/main.js [hunk 17]
```diff
@@ -14,4 +16,137 @@ menuButton.forEach((item) => {
...
+const openCloseModal = () => {
+  modalSection.classList.toggle('modal');
+  modalSection.classList.toggle('modal-init');
+};
+
+// Add Event to project buttons
```
Comment (vichuge) [inline scripts/main.js:145]: - [ ] A good practice on code, is to delete all the commented lines, I kindly invite you to delete it. 🐶

File: scripts/main.js [hunk 18]
```diff
@@ -14,4 +16,137 @@ menuButton.forEach((item) => {
...
+projectButton.forEach((item) => {
+  item.addEventListener('click', openCloseModal);
+});
+
+// Add Event to close icon button
```
Comment (vichuge) [inline scripts/main.js:150]: - [ ] A good practice on code, is to delete all the commented lines, I kindly invite you to delete it. 🐶

File: styles/main.css [hunk 19]
```diff
@@ -91,7 +91,7 @@ html {
 /* Headline */
```
Comment (vichuge) [inline styles/main.css:91]: - [ ] A good practice on code, is to delete all the commented lines, I kindly invite you to delete it. 🐶

File: styles/main.css [hunk 20]
```diff
@@ -271,7 +271,7 @@ html {
 /* About */
```
Comment (vichuge) [inline styles/main.css:271]: - [ ] A good practice on code, is to delete all the commented lines, I kindly invite you to delete it. 🐶

File: styles/main.css [hunk 21]
```diff
@@ -374,7 +374,7 @@ html {
 /* Contact Form */
```
Comment (vichuge) [inline styles/main.css:374]: - [ ] A good practice on code, is to delete all the commented lines, I kindly invite you to delete it. 🐶

File: styles/main.css [hunk 22]
```diff
@@ -461,6 +461,114 @@ html {
   margin: 25px 0 50px 0;
 }
 
+/* Modal */
```
Comment (vichuge) [inline styles/main.css:464]: - [ ] A good practice on code, is to delete all the commented lines, I kindly invite you to delete it. 🐶

File: styles/main.css [hunk 23]
```diff
@@ -461,6 +461,114 @@ html {
...
+  font-weight: 500;
+}
+
+.modal-btn a {
+  text-decoration: none;
+  color: var(--light-color);
+}
+
 @media (min-width: 768px) {
   /* Navigation */
```
Comment (vichuge) [inline styles/main.css:573]: - [ ] A good practice on code, is to delete all the commented lines, I kindly invite you to delete it. 🐶

File: styles/main.css [hunk 24]
```diff
@@ -494,7 +602,7 @@ html {
 
   /* Headline */
```
Comment (vichuge) [inline styles/main.css:603]: - [ ] A good practice on code, is to delete all the commented lines, I kindly invite you to delete it. 🐶

File: styles/main.css [hunk 25]
```diff
@@ -595,6 +703,87 @@ html {
   .footer-email {
     display: none;
   }
+
+  /* Modal */
```
Comment (vichuge) [inline styles/main.css:707]: - [ ] A good practice on code, is to delete all the commented lines, I kindly invite you to delete it. 🐶

**Review by vichuge — CHANGES_REQUESTED** [review]
# Required Changes ♻️ ⚠️ 🚦 

Hi dear student, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

mario_ladders

## Good points 👍 
- No Linters error ✔️
- I like this project a lot ✔️ 

## Aspects to improve ♻️

- [ ] Check the comments under the review (I know, commented lines could be important to the code understanding, at the same time to keep the code clean could help to have a better work on code, talking about comments, we could use tools inside GitHub like branches, commits and others tools to have a great code organization).

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have only 3 code reviews per project. If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by arslanbisharat — CHANGES_REQUESTED** [review]
# Approved 👏 

Hi @alonsofl, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have only 3 code reviews per project. If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)_

**Pull request merged**